### PR TITLE
make\arm: Fix compiling against newlib nano.specs

### PR DESCRIPTION
**Pull request by DipSwitch**
When linking against newlib-nano it is a good idea to make the compiler include the proper newlib configuration header.

Needed for: #4529

### CODE REVIEW

**Comment by DipSwitch** [comment]
This fix would reduce the size of `struct _reent` with 968 bytes on ARM platforms.

Before and after:

	main(): This is RIOT! (Version: 2016.03-devel-1105-g9e02-dipswitch-K55VD-pr/add_newlib_threadsafe)
			member, sizeof, offsetof
	sizeof(thread_t): 1104
			sp            4   0
			status        1   4
			priority      1   5
			pid           2   6
			rq_entry      8   8
			wait_data     4  16
			msg_waiters   4  20
			msg_queue    12  24
			msg_array     4  36
			newlib_reent1064  40
	Done.


	main(): This is RIOT! (Version: 2016.03-devel-1105-g9e02-dipswitch-K55VD-pr/add_newlib_threadsafe)
			member, sizeof, offsetof
	sizeof(thread_t): 136
			sp            4   0
			status        1   4
			priority      1   5
			pid           2   6
			rq_entry      8   8
			wait_data     4  16
			msg_waiters   4  20
			msg_queue    12  24
			msg_array     4  36
			newlib_reent  96  40
	Done.

File: cpu/Makefile.include.cortexm_common [hunk 1]
```diff
@@ -121,6 +121,8 @@ include $(RIOTCPU)/Makefile.include.$(TOOLCHAIN)
 
 # use the nano-specs of Newlib when available
 ifeq ($(shell $(LINK) -specs=nano.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+export CFLAGS += -specs=nano.specs
+export CXXFLAGS += -specs=nano.specs
```
Comment (gebart) [inline cpu/Makefile.include.cortexm_common]: CFLAGS is already included in CXXFLAGS by Makefile.include

**Comment by DipSwitch** [comment]
Changed and squashed

**Comment by gebart** [comment]
I just realized that boards/msba2-common/Makefile.include also contains a similar line, could you update that one as well?

**Comment by gebart** [comment]
ACK once the above mentioned change is made.

**Comment by DipSwitch** [comment]
Addressed, and squashed

**Comment by DipSwitch** [comment]
I'll aim for RIOT-CI and skill travis on this PR. :)

**Comment by DipSwitch** [comment]
Fails on RIOT-CI, while when I didn't change the msba2 make script It passed.

But it maybe has to do with the build command is having double defined parameters:

arm-none-eabi-gcc -o lib-ccn-lite.a -DDEVELHELP -DUSE_LINKLAYER -DUSE_IPV6 -DCCNL_UAPI_H_ -DUSE_SUITE_NDNTLV -DNEEDS_PREFIX_MATCHING -DNEEDS_PACKET_CRAFTING -DUSE_HMAC256 -Werror -DFEATURE_PERIPH_TIMER -DMODULE_ARM7_COMMON -DMODULE_AUTO_INIT -DMODULE_AUTO_INIT_GNRC_NETIF -DMODULE_BITFIELD -DMODULE_BOARD -DMODULE_CC110X -DMODULE_CCN_LITE -DMODULE_CCN_LITE_UTILS -DMODULE_CORE -DMODULE_CPU -DMODULE_GNRC -DMODULE_GNRC_CC110X -DMODULE_GNRC_NETAPI -DMODULE_GNRC_NETDEV2 -DMODULE_GNRC_NETIF -DMODULE_GNRC_NETIF_DEFAULT -DMODULE_GNRC_NETIF_HDR -DMODULE_GNRC_NETREG -DMODULE_GNRC_PKT -DMODULE_GNRC_PKTBUF -DMODULE_GNRC_PKTBUF_STATIC -DMODULE_IEEE802154 -DMODULE_MSBA2_COMMON -DMODULE_MSBA2_COMMON_DRIVERS -DMODULE_NETOPT -DMODULE_NEWLIB -DMODULE_NEWLIB_SYSCALLS_DEFAULT -DMODULE_PERIPH -DMODULE_PERIPH_COMMON -DMODULE_PRNG_MINSTD -DMODULE_PS -DMODULE_RANDOM -DMODULE_SHELL -DMODULE_SHELL_COMMANDS -DMODULE_SYS -DMODULE_TIMEX -DMODULE_TLSF -DMODULE_TSRB -DMODULE_UART_STDIO -DMODULE_XTIMER `-mcpu=arm7tdmi-s -ffunction-sections -fdata-sections -fno-builtin -fshort-enums -ggdb -g3 -Os -specs=nano.specs -DBOARD_MSBA2=\"msba2\" -DRIOT_BOARD=BOARD_MSBA2 -DCPU_LPC2387=\"lpc2387\" -DRIOT_CPU=CPU_LPC2387 -DMCU_LPC2387=\"lpc2387\" -DRIOT_MCU=MCU_LPC2387 -std=gnu99 -fno-delete-null-pointer-checks -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition -fno-common -Wall` -DRIOT_VERSION=\"\" `-mcpu=arm7tdmi-s -ffunction-sections -fdata-sections -fno-builtin -fshort-enums -ggdb -g3 -Os -specs=nano.specs`  -isystem /etc/alternatives/gcc-arm-none-eabi-include -I/data/riotbuild/core/include -I/data/riotbuild/drivers/include -I/data/riotbuild/sys/include -I/data/riotbuild/cpu/lpc2387/include -I/data/riotbuild/boards/msba2/include -I/data/riotbuild/boards/msba2-common/include -I/data/riotbuild/boards/msba2-common/drivers/include -I/data/riotbuild/cpu/arm7_common/include/ -I/data/riotbuild/sys/libc/include -I/data/riotbuild/drivers/cc110x/include -I/data/riotbuild/pkg_arm7/tlsf/tlsf -I/data/riotbuild/pkg_arm7/ccn-lite -I/data/riotbuild/pkg_arm7/ccn-lite/ccn-lite/src -I/data/riotbuild/sys/posix/include -I/data/riotbuild/boards/msba2-common/include -I/data/riotbuild/boards/msba2-common/drivers/include -Wall -Werror -std=c99 -g -D_XOPEN_SOURCE=500 -D_XOPEN_SOURCE_EXTENDED -Dlinux -Wno-date-time -c ccn-lite-riot.c

**Comment by DipSwitch** [comment]
But when building CCN nano.specs isn't passed to to the build process of CCN.

**Comment by DipSwitch** [comment]
Seems that there `CFLAGS` is always appended twice to a gcc command for `ARM7` CPU's and `smba2` is also used for the lpc2387. Searching where this is coming from.

**Comment by DipSwitch** [comment]
For the `avsextreme` is found.

**Comment by DipSwitch** [comment]
@OlegHahm Does this hurt CCN-LITE? It seems the  build command is still valid and has all the defines.

	arm-none-eabi-gcc -o lib-ccn-lite.a -DDEVELHELP -DUSE_LINKLAYER -DUSE_IPV6 -DCCNL_UAPI_H_ -DUSE_SUITE_NDNTLV -DNEEDS_PREFIX_MATCHING -DNEEDS_PACKET_CRAFTING -DUSE_HMAC256 -Werror -DFEATURE_PERIPH_TIMER -DMODULE_ARM7_COMMON -DMODULE_AUTO_INIT -DMODULE_AUTO_INIT_GNRC_NETIF -DMODULE_BITFIELD -DMODULE_BOARD -DMODULE_CC110X -DMODULE_CCN_LITE -DMODULE_CCN_LITE_UTILS -DMODULE_CORE -DMODULE_CPU -DMODULE_GNRC -DMODULE_GNRC_CC110X -DMODULE_GNRC_NETAPI -DMODULE_GNRC_NETDEV2 -DMODULE_GNRC_NETIF -DMODULE_GNRC_NETIF_DEFAULT -DMODULE_GNRC_NETIF_HDR -DMODULE_GNRC_NETREG -DMODULE_GNRC_PKT -DMODULE_GNRC_PKTBUF -DMODULE_GNRC_PKTBUF_STATIC -DMODULE_IEEE802154 -DMODULE_MSBA2_COMMON -DMODULE_MSBA2_COMMON_DRIVERS -DMODULE_NETOPT -DMODULE_NEWLIB -DMODULE_NEWLIB_SYSCALLS_DEFAULT -DMODULE_PERIPH -DMODULE_PERIPH_COMMON -DMODULE_PRNG_MINSTD -DMODULE_PS -DMODULE_RANDOM -DMODULE_SHELL -DMODULE_SHELL_COMMANDS -DMODULE_SYS -DMODULE_TIMEX -DMODULE_TLSF -DMODULE_TSRB -DMODULE_UART_STDIO -DMODULE_XTIMER -mcpu=arm7tdmi-s -ffunction-sections -fdata-sections -fno-builtin -fshort-enums -ggdb -g3 -Os -specs=nano.specs -DBOARD_MSBA2=\"msba2\" -DRIOT_BOARD=BOARD_MSBA2 -DCPU_LPC2387=\"lpc2387\" -DRIOT_CPU=CPU_LPC2387 -DMCU_LPC2387=\"lpc2387\" -DRIOT_MCU=MCU_LPC2387 -std=gnu99 -fno-delete-null-pointer-checks -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition -fno-common -Wall -DRIOT_VERSION=\"2016.03-devel-1011-g12c31-dipswitch-K55VD-pr/fix_arm_newlib_nano_specs\"  -isystem /etc/alternatives/gcc-arm-none-eabi-include -I/home/dipswitch/RIOT/core/include -I/home/dipswitch/RIOT/drivers/include -I/home/dipswitch/RIOT/sys/include -I/home/dipswitch/RIOT/cpu/lpc2387/include -I/home/dipswitch/RIOT/boards/msba2/include -I/home/dipswitch/RIOT/boards/msba2-common/include -I/home/dipswitch/RIOT/boards/msba2-common/drivers/include -I/home/dipswitch/RIOT/cpu/arm7_common/include/ -I/home/dipswitch/RIOT/sys/libc/include -I/home/dipswitch/RIOT/drivers/cc110x/include -I/home/dipswitch/RIOT/pkg/tlsf/tlsf -I/home/dipswitch/RIOT/pkg/ccn-lite -I/home/dipswitch/RIOT/pkg/ccn-lite/ccn-lite/src -I/home/dipswitch/RIOT/sys/posix/include -Wall -Werror -std=c99 -g -D_XOPEN_SOURCE=500 -D_XOPEN_SOURCE_EXTENDED -Dlinux -Wno-date-time -c ccn-lite-riot.c
	"make" -BC /home/dipswitch/RIOT/pkg/ccn-lite/ccn-lite/src lib-ccn-lite-utils.a

@kaspar030 Does this hurt the MBSA2 common build?

It seems this same "mistake" is made in all packages. Maybe this is due to Makefile changes later on. But probably all below would break if I want to add `-specs=nano.specs` to the CFLAGS.

	0) dipswitch@dipswitch-K55VD:~/RIOT $ grep -R '$(RIOTBOARD)/$(BOARD)/Makefile.include' *
	boards/wsn430-common/drivers/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	boards/msb-430-common/drivers/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	boards/msba2-common/drivers/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	Makefile.include:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/oonf_api/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/relic/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/micro-ecc/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/microcoap/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/openwsn/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/libcoap/Makefile:include $(RIOTBOARD)/$(BOARD)/Makefile.include
	pkg/Makefile.git:include $(RIOTBOARD)/$(BOARD)/Makefile.include

**Comment by OlegHahm** [comment]
@DipSwitch, I will give a try.

**Comment by OlegHahm** [comment]
Seems to be okay.

**Comment by gebart** [comment]
ACK, could you tidy up the commit list a bit?
Squash all the stray board Makefile.include includes into one commit and put it before the newlib-nano commit (so 2 commits total)

**Comment by DipSwitch** [comment]
Rebased and squashed

**Comment by DipSwitch** [comment]
Unlocks #4529 after merge

**Comment by gebart** [comment]
Merge at will when Murdock is back online.

**Comment by DipSwitch** [comment]
Or if travis is green before Murdock arrives :p

**Comment by gebart** [comment]
nhdp is failing because of #5107. Merging anyway.

**Pull request merged**